feat: create workflow page (#3794)
notifirehq/notifire: docs/docs/platform/workflows/create-workflow.md

---
sidebar_position: 1
sidebar_label: Create Workflow
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

# Create Workflow

> We have an elegant drag and drop interface to create a new `workfklow`. User can wither create new workflow from scratch, by selecting existing workflow from our workflow store or programmatically using API/SDK. Click on the `Workflows` option in the left sidebar to access all workflows. Workflows page lists all exisiting workflows ordered by last creation date.

Reviewer: ```suggestion
We have an elegant drag and drop interface to create a new `workflow`. User can either create a new workflow from scratch, select an existing workflow from our Template Store or create one programmatically using API/SDK. Click on the `Workflows` option in the left sidebar to access all workflows. Workflows page lists all existing workflows ordered by last creation date.
```
Note, that we call it Template Store , also it would be nice to have a link to template store

## Create a new workflow from scratch

> On `/workflows` page, click on the `Create Workflow` button. Choose `Blank Workflow` option to create a new workflow from scratch. Add `channel` and `action` nodes either by drag and drop from right sidebar or by clicking on the `plus icon` at terminal or in-between of nodes.

Reviewer: at terminal - you mean at the plus icon after the last node?

<div align="center">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="/img/platform/workflow/create-new-workflow-from-blank.gif"/>
    <img src="/img/platform/workflow/create-new-workflow-from-blank.gif" alt="create-new-workflow-from-blank"/>
  </picture>
</div>

> ## Using pre-made workflows from workflow store

Reviewer: ```suggestion
## Using pre-made workflows from the template store
```

> On `/workflows` page, click on the `Create Workflow` button. There will be list of pre-made workflows created by Novu team. Choose any one option to create a new workflow. This workflow will have steps and steps will already have content based on the workflow type. Any new `channel` and `action` nodes can be added either by drag and drop from right sidebar or by clicking on the `plus icon` at terminal or in-between of nodes.

Reviewer: ```suggestion
On `/workflows` page, click on the `Create Workflow` button. There will be list of pre-made workflow templates created by Novu team. Choose any option to create a new workflow with pre-configured steps and content based on the workflow type. Any new `channel` and `action` nodes can be added either by drag and drop from right sidebar or by clicking on the `plus icon`.
```

      filters: [
        {
          value: 'AND',
          children: [
            {
              field: '{{chatContent}}',
              value: 'flag',
              operator: 'NOT_IN',
              // 'payload'
              on: FilterPartTypeEnum.PAYLOAD,
            },
          ],
        },
      ],
      template: {
        // 'chat'
        type: StepTypeEnum.CHAT,
        active: true,
        subject: '',
        variables: [
          {
            name: 'chatContent',
            // 'String'
            type: TemplateVariableTypeEnum.STRING,
            required: true,
          },
        ],
        content: '{{chatContent}}',
        contentType: 'editor',
      },
    },
  ],
  description: 'Onboarding workflow to trigger after user sign up',
  active: true,
  draft: false,
  critical: false,
});
```

  </TabItem>
  <TabItem value="curl" label="Curl">

```shell
curl --location 'https://api.novu.co/v1/notification-templates' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: ApiKey <NOVU_API_KEY>' \
--data '{
  "name": "Onboarding Workflow",
  "notificationGroupId": "644412354062e17c2e21b5fe",

>   "steps": [],

Reviewer: The first example is with creating a step, maybe we could show it here as well?
